packages/ui/src/ui/pages/query-tracker/module/query/actions.ts

```diff
@@ -138,6 +109,44 @@ export const loadCliqueByCluster =
             });
     };
 
+export function loadQuery(
+    queryId: string,
+): ThunkAction<any, RootState, any, SetQueryAction | RequestQueryAction | SetQueryErrorLoadAction> {
+    return async (dispatch, _getState) => {
+        dispatch({type: REQUEST_QUERY});
+        try {
+            const query = await wrapApiPromiseByToaster(dispatch(getQuery(queryId)), {
+                toasterName: 'load_query',
+                skipSuccessToast: true,
+                errorTitle: 'Failed to load query',
+            });
+
+            if (
+                query.engine === QueryEngine.CHYT &&
+                query.settings &&
```

Comment:
If we describe type of query.settings as `{cluster?: string} & Record<string, unknown>` we can use here:
```ts
if (query.engine === QueryEngine.CHYT && query.settings?.cluster) {
   ...
}
```